src/main/java/com/thealgorithms/maths/AutoCorrelation.java

```diff
@@ -0,0 +1,84 @@
+package com.thealgorithms.maths;
+
+
+/**
+ * Class for linear auto-correlation of a discrete signal
+ *
+ * @author Athina-Frederiki Swinkels
+ * @version 1.0
+ */
+
+public class AutoCorrelation {
+
+    /**
+     * Discrete linear auto-correlation function.
+     * Input and output signals have starting index 0.
+     *
+     * @param x The discrete signal
+     * @return The result of the auto-correlation of signals x. The result is also a signal.
+     */
+    public static double[] autoCorrelation(double[] x){
+        //The result signal's length is double the length of signal x minus 1
+        double[] result=new double[2*x.length-1];
+        int N=result.length;
+
+        /*
+        To find the auto-correlation of a discrete signal x, we perform cross-correlation between x signal and itself.
```

Comment:
Could you just call `crossCorrelation(x, x)` then?

Reply:
Ok, I was debating between choosing class autonomy or least amount of code lines. If class autonomy is not neccessary then I will do it this way

Reply:
Ok, I was debating between choosing class autonomy or least amount of code lines. If class autonomy is not neccessary than I will do it this way